### ISSUE TITLE
[Feature] "Quick Action" shortcut for toolbar actions

### ISSUE THREAD

**Issue by ZaLiTHkA** [sourcegit-scm/sourcegit#351]
tl;dr: I'm a long-time Git Fork user who switched over to SourceGit after migrating my dev environment from Windows to Linux (we've been waiting for a linux version for years, will probably wait a few more years for that).

one feature of Fork that I really do miss, is the ability to run a "Fetch" or "Pull" or "Push" with a <kbd>ctrl</kbd>+click on the relevant toolbar button. so where a standard click on "Fetch" will currently open a popup to allow the fetch action to be configured, holding <kbd>ctrl</kbd> while clicking would instead just run fetch with all the default options.

any chance we could get this functionality in SourceGit as well? it may seem like an insignificant feature, but it does make a pretty big difference to the "user experience" side of things.

**Comment by love-linger** [comment]
you can test this feature from https://github.com/sourcegit-scm/sourcegit/actions/runs/10347338234

**Comment by gadfly3173** [comment]
This feature did not work on my Deepin V23. Only when I press the `Ctrl` key after click the toolbar button, then `OnKeyDown` would be called.

**Comment by ZaLiTHkA** [comment]
> you can test this feature from [sourcegit-scm/sourcegit/actions/runs/10347338234](https://github.com/sourcegit-scm/sourcegit/actions/runs/10347338234)

you're too quick with these things... you're gonna make other OSS project maintainers look bad. (: lol.

just busy finishing up some time critical tasks at the moment, I'll test this one Windows 11 Pro shortly.

**Comment by love-linger** [comment]
> This feature did not work on my Deepin V23. Only when I press the `Ctrl` key after click the toolbar button, then `OnKeyDown` would be called.

It's hard to get the key modifiers without `KeyDown` event. Unfortunately, Avalonia Team doesn't seem to have the will to change that. See https://github.com/AvaloniaUI/Avalonia/issues/3318

**Comment by love-linger** [comment]
@gadfly3173 I've pushed a new commit that try to fix this issue, would please test it for me?

**Comment by ZaLiTHkA** [comment]
just [this build](https://github.com/sourcegit-scm/sourcegit/actions/runs/10348286270 on my side. "Fetch", "Push" and "Pull" all seem to work just fine for me.

not sure if the first test build had the same issue on Windows, but the second one is good.

**Comment by gadfly3173** [comment]
> @gadfly3173 I've pushed a new commit that try to fix this issue, would please test it for me?我已经推送了一个新的提交来尝试解决这个问题，请为我测试一下吗？

I still can't use that. Maybe `e.KeyModifiers` is wrong.
![1723451929534](https://github.com/user-attachments/assets/c7541b64-2195-4dd5-bab7-41ac6b273435)

**Comment by love-linger** [comment]
I think I know how to fix this. 
![image](https://github.com/user-attachments/assets/6b3307f0-40f1-4184-a97d-e2cd77e3e88e)

**Comment by love-linger** [comment]
Looking forward for your feedback @gadfly3173

**Comment by gadfly3173** [comment]
> Looking forward for your feedback @gadfly3173

It works fine now.

**Comment by ZaLiTHkA** [comment]
thanks @love-linger for the super quick turn-around, and thanks to @gadfly3173 for testing the Linux side. 🍻
